Extensions/retags.js

```diff
@@ -20,16 +20,24 @@ XKit.extensions.retags = {
 		} catch(e) {}
 		this.add_toggle();
 		this.observer.observe($('body')[0],{childList:true,subtree:true});
-		this.tag(this.selectors);
+		XKit.extensions.retags.tag(XKit.extensions.retags.selectors);
 	},
 
 	observer: new MutationObserver(function(ms){
-		var inner, $el;
+		var inner, innerFallback, $el;
+		var notesProcessed = false;
 		ms.forEach(function(m){
 			$el = $(m.addedNodes);
-			inner = $el.find('.note:not(.ui_post_badge)');
-			if (inner.length > 0) {
-				XKit.extensions.retags.tag(inner.filter(XKit.extensions.retags.selectors));
+			var nonPopoverNotes = $el.find('.note:not(.ui_post_badge)');
+
+			if ($el.hasClass('note-list')) {
```

Comment:
@nightpool I haven't had a chance to look at this again since life got crazy again (watching calendar for June), but I have been using this extension locally on my machine since I submitted the PR and haven't noticed any performance degradation, FWIW.